install-dev.sh install bug on Initialized AWS Ubuntu Instances (ex. 20.04.03 LTS)

**Describe the bug**
A clear and concise description of what the bug is.
문제가 발생하는 상황에 대한 요약 설명을 적어주세요.

When I install backend.ai using install-dev.sh on AWS Ubuntu.
Because of not to install `build-essential`, doesn't install `backend-ai` using `install-dev.sh`.

example)

Ubuntu 20.04.03 LTS
```
[INFO] Installing Python...
Downloading Python-3.9.9.tar.xz...
-> https://www.python.org/ftp/python/3.9.9/Python-3.9.9.tar.xz
Installing Python-3.9.9...

BUILD FAILED (Ubuntu 20.04 using python-build 20180424)

Inspect or clean up the working tree at /tmp/python-build.20220209083139.23229
Results logged to /tmp/python-build.20220209083139.23229.log

Last 10 log lines:
checking for python3... python3
checking for --enable-universalsdk... no
checking for --with-universal-archs... no
checking MACHDEP... "linux"
checking for gcc... no
checking for cc... no
checking for cl.exe... no
configure: error: in `/tmp/python-build.20220209083139.23229/Python-3.9.9':
configure: error: no acceptable C compiler found in $PATH
See `config.log' for more details
 
[ERROR] Installing the Python version 3.9.9 via pyenv has failed.
 
[NOTE] 3.9.9 is not supported by your current installation of pyenv.
 
[NOTE] Please update pyenv or lower PYTHON_VERSION in install-dev.sh script.
ubuntu@ip-10-122-3-142:~$ python --version

Command 'python' not found, did you mean:

  command 'python3' from deb python3
  command 'python' from deb python-is-python3

ubuntu@ip-10-122-3-142:~$ uname -a
Linux ip-10-122-3-142 5.11.0-1020-aws #21~20.04.2-Ubuntu SMP Fri Oct 1 13:03:59 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
ubuntu@ip-10-122-3-142:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04
Codename:	focal
ubuntu@ip-10-122-3-142:~$ 
```





**To Reproduce**
Steps to reproduce the behavior:
문제를 재현하기 위한 순서를 자세히 적어주세요:
1. download `install-dev.sh`
2. Run `install-dev.sh`
3. See error

Ubuntu 20.04.03 LTS
```
[INFO] Installing Python...
Downloading Python-3.9.9.tar.xz...
-> https://www.python.org/ftp/python/3.9.9/Python-3.9.9.tar.xz
Installing Python-3.9.9...

BUILD FAILED (Ubuntu 20.04 using python-build 20180424)

Inspect or clean up the working tree at /tmp/python-build.20220209083139.23229
Results logged to /tmp/python-build.20220209083139.23229.log

Last 10 log lines:
checking for python3... python3
checking for --enable-universalsdk... no
checking for --with-universal-archs... no
checking MACHDEP... "linux"
checking for gcc... no
checking for cc... no
checking for cl.exe... no
configure: error: in `/tmp/python-build.20220209083139.23229/Python-3.9.9':
configure: error: no acceptable C compiler found in $PATH
See `config.log' for more details
 
[ERROR] Installing the Python version 3.9.9 via pyenv has failed.
 
[NOTE] 3.9.9 is not supported by your current installation of pyenv.
 
[NOTE] Please update pyenv or lower PYTHON_VERSION in install-dev.sh script.
ubuntu@ip-10-122-3-142:~$ python --version

Command 'python' not found, did you mean:

  command 'python3' from deb python3
  command 'python' from deb python-is-python3

ubuntu@ip-10-122-3-142:~$ uname -a
Linux ip-10-122-3-142 5.11.0-1020-aws #21~20.04.2-Ubuntu SMP Fri Oct 1 13:03:59 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
ubuntu@ip-10-122-3-142:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04
Codename:	focal
ubuntu@ip-10-122-3-142:~$ 
```


**Expected behavior**
A clear and concise description of what you expected to happen.
원래 기대했던 동작은 어떤 것인지 명확하게 적어주세요.

Successfully install backend.ai on Ubuntu 

**Screenshots**
If applicable, add screenshots to help explain your problem.
Just take a screenshot in your clipboard or copy a picture file and paste it here (Ctrl+V/Cmd+V) so that GitHub automatically attach and insert it.
가능하다면, 문제 상황을 보여주는 스크린샷을 첨부해주세요.
클립보드에 스크린샷을 찍거나 이미지 파일을 복사해서 여기에 붙여넣으면(Ctrl+V/Cmd+V) GitHub이 자동으로 이곳에 첨부 및 삽입해줍니다.



**Server:** 
 - OS: Ubuntu 20.04.3 LTS
   - Output of `uname -a`: 
 `Linux ip-10-122-3-142 5.11.0-1020-aws #21~20.04.2-Ubuntu SMP Fri Oct 1 13:03:59 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`
   -  Backend.AI version:  `main`
 - Python version:  None 
 - Installation method: `used scripts/install-dev.sh`
<img width="886" alt="image" src="https://user-images.githubusercontent.com/20002/153163957-2e394e80-54a7-4d79-93b3-e95bf50c79fc.png">



**Additional context**
Add any other context about the problem here.
추가적으로 문제 상황을 설명할 수 있는 다른 맥락을 기술해주세요.

I think that add lines on `install-dev.sh`
1. Ubuntu
  * add `build-essential` package
  https://github.com/lablup/backend.ai/blob/main/scripts/install-dev.sh#L263

on CentOS
After `yum groupinstall -y "Development Tools"`, try to install backend.ai using `install-dev.sh`, shows error message

![image](https://user-images.githubusercontent.com/20002/153339806-507bc8f0-ee79-4003-9b71-472db5df625a.png)

If you are using centos and compile python from source, you can install from following commands
For centos: `sudo yum install -y xz-devel`
```
[INFO] Installing Python...
Downloading Python-3.9.9.tar.xz...
-> https://www.python.org/ftp/python/3.9.9/Python-3.9.9.tar.xz
Installing Python-3.9.9...
     WARNING: The Python sqlite3 extension was not compiled. Missing the SQLite3 lib?
Installed Python-3.9.9 to /home/centos/.pyenv/versions/3.9.9

 
[INFO] Checking Python features...
SSL support: ok
 
[ERROR] Your Python (prefix: /home/centos/.pyenv/versions/3.9.9) is missing LZMA (XZ) support. Please reinstall or rebuild it.
```
Ref
https://stackoverflow.com/a/60890502

--- a/scripts/install-dev.sh
+++ b/scripts/install-dev.sh
@@ -260,11 +260,11 @@ install_script_deps() {
   case $DISTRO in
   Debian)
     $sudo apt-get update
-    $sudo apt-get install -y git jq
+    $sudo apt-get install -y git jq gcc make
     ;;
   RedHat)
     $sudo yum clean expire-cache  # next yum invocation will update package metadata cache
-    $sudo yum install -y git jq
+    $sudo yum install -y git jq gcc make
     ;;
   Darwin)
     if ! type "brew" >/dev/null 2>&1; then
@@ -285,7 +285,7 @@ install_pybuild_deps() {
     $sudo apt-get install -y libssl-dev libreadline-dev libgdbm-dev zlib1g-dev libbz2-dev libsqlite3-dev libffi-dev liblzma-dev
     ;;
   RedHat)
-    $sudo yum install -y openssl-devel readline-devel gdbm-devel zlib-devel bzip2-devel libsqlite-devel libffi-devel lzma-devel
+    $sudo yum install -y openssl-devel readline-devel gdbm-devel zlib-devel bzip2-devel libsqlite-devel libffi-devel lzma-devel xz-devel
     ;;
   Darwin)
     brew install openssl

